subsys/bluetooth/services/ias.c

```diff
@@ -0,0 +1,122 @@
+/** @file
+ *  @brief IAS Service sample
+ */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "net/buf.h"
+#include "bluetooth/services/ias.h"
+#include <init.h>
+#include <zephyr.h>
+#include <logging/log.h>
+#include <bluetooth/gatt.h>
+
+#define LOG_LEVEL CONFIG_BT_IAS_LOG_LEVEL
+LOG_MODULE_REGISTER(ias);
+
+#define ALERT_MAX_LEN 1
+#define BT_IAS_ALERT_LVL_NO_ALERT		0
+#define BT_IAS_ALERT_LVL_MILD_ALERT		1
+#define BT_IAS_ALERT_LVL_HIGH_ALERT		2
+
+#define GATT_PERM_WRITE_MASK    (BT_GATT_PERM_WRITE | \
+				 BT_GATT_PERM_WRITE_ENCRYPT | \
+				 BT_GATT_PERM_WRITE_AUTHEN)
+
+#ifndef CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN
+#define CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN 0
+#endif
+#ifndef CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT
+#define CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT 0
+#endif
+
+#define IAS_GATT_PERM_DEFAULT (				\
+	CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN ?	\
+	(BT_GATT_PERM_WRITE_AUTHEN) :			\
+	CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT ?	\
+	(BT_GATT_PERM_WRITE_ENCRYPT) :			\
+	(BT_GATT_PERM_WRITE))					\
+
+static void alert_stop()
+{
+	LOG_INF("Alert stopped");
+}
+
+static void alert_start()
+{
+	LOG_INF("Mild alert started");
+}
+
+static void alert_high_start()
+{
+	LOG_INF("High alert started");
+}
+
+BT_IAS_CB_DEFINE(ias_callbacks) = {
+	.start_alert = alert_start,
+	.stop_alert = alert_stop,
+	.start_alert_high = alert_high_start,
+};
```

Comment:
What is the reasoning behind using the service callbacks to log (with info) in the service itself? 
1) We can save some ROM by not declaring `BT_IAS_CB_DEFINE` and instead call them directly
2) Since the service itself do not use the values for anything, I suggest to use LOG_DBG. 

Alternatively remove all of this, and simply log directly in the `alert_level` function.